platform/core/src/services/DisplaySetService/DisplaySetService.ts

```diff
@@ -236,34 +269,70 @@ export default class DisplaySetService extends PubSubService {
         );
 
         if (displaySets.length) {
-          this._addActiveDisplaySets(displaySets);
-        } else {
-          displaySets = handler.getDisplaySetsFromSeries(instances);
+          instances = filterInstances(instances, displaySets);
+          for (const ds of displaySets) {
+            const addedDs = ds.addInstances?.(instances, this);
+            if (addedDs) {
+              this.activeChanged = true;
+              instances = filterInstances(instances, [ds]);
+              this._addActiveDisplaySets([ds]);
+              this.setDisplaySetMetadataInvalidated(ds.displaySetInstanceUID);
+            }
+            if (!instances.length) return allDisplaySets;
+          }
+          if (instances.length === 0) {
+            // Everything is already added - this is just an update caused
+            // by something else
+            this._addActiveDisplaySets(displaySets);
+            return allDisplaySets;
+          }
+          // There are  still instances to add, and they will get added as
+          // a new display set.
```

Comment:
and which instances are they? the ones that don't have the ds created (so we can't "append" them) right? please add that detail

Reply:
and it should moved one line down for the rest of the logic